dlt-daemon Compilation error for release v2.17.0

gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.6)

Fix: uncomment line 343 in include/dlt/dlt_common.h
```
/**
 * Provision to test static function
 */
#ifndef DLT_UNIT_TESTS
//#define STATIC static // <-- UNCOMMENT ME
#else
#define STATIC
#endif
```


```
dlt-daemon/build> make 
[  0%] Built target compress_man
[  8%] Built target dlt
[  9%] Building C object src/daemon/CMakeFiles/dlt-daemon.dir/dlt_daemon_connection.c.o
/home/vagrant/dlt-daemon/src/daemon/dlt_daemon_connection.c:66:8: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘int’
 STATIC int dlt_connection_send(DltConnection *conn,
        ^
/home/vagrant/dlt-daemon/src/daemon/dlt_daemon_connection.c: In function ‘dlt_connection_send_multiple’:
/home/vagrant/dlt-daemon/src/daemon/dlt_daemon_connection.c:118:15: warning: implicit declaration of function ‘dlt_connection_send’ [-Wimplicit-function-declaration]
         ret = dlt_connection_send(con,
               ^
/home/vagrant/dlt-daemon/src/daemon/dlt_daemon_connection.c: At top level:
/home/vagrant/dlt-daemon/src/daemon/dlt_daemon_connection.c:163:8: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘void’
 STATIC void dlt_connection_destroy_receiver(DltConnection *con)
        ^
/home/vagrant/dlt-daemon/src/daemon/dlt_daemon_connection.c:195:1: error: unknown type name ‘STATIC’
 STATIC DltReceiver *dlt_connection_get_receiver(DltDaemonLocal *daemon_local,
 ^
/home/vagrant/dlt-daemon/src/daemon/dlt_daemon_connection.c:195:20: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘*’ token
 STATIC DltReceiver *dlt_connection_get_receiver(DltDaemonLocal *daemon_local,
                    ^
/home/vagrant/dlt-daemon/src/daemon/dlt_daemon_connection.c: In function ‘dlt_connection_destroy’:
/home/vagrant/dlt-daemon/src/daemon/dlt_daemon_connection.c:326:5: warning: implicit declaration of function ‘dlt_connection_destroy_receiver’ [-Wimplicit-function-declaration]
     dlt_connection_destroy_receiver(to_destroy);
     ^
/home/vagrant/dlt-daemon/src/daemon/dlt_daemon_connection.c: In function ‘dlt_connection_create’:
/home/vagrant/dlt-daemon/src/daemon/dlt_daemon_connection.c:381:22: warning: implicit declaration of function ‘dlt_connection_get_receiver’ [-Wimplicit-function-declaration]
     temp->receiver = dlt_connection_get_receiver(daemon_local, type, fd);
                      ^
/home/vagrant/dlt-daemon/src/daemon/dlt_daemon_connection.c:381:20: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
     temp->receiver = dlt_connection_get_receiver(daemon_local, type, fd);
                    ^
src/daemon/CMakeFiles/dlt-daemon.dir/build.make:110: recipe for target 'src/daemon/CMakeFiles/dlt-daemon.dir/dlt_daemon_connection.c.o' failed
make[2]: *** [src/daemon/CMakeFiles/dlt-daemon.dir/dlt_daemon_connection.c.o] Error 1
CMakeFiles/Makefile2:258: recipe for target 'src/daemon/CMakeFiles/dlt-daemon.dir/all' failed
make[1]: *** [src/daemon/CMakeFiles/dlt-daemon.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/include/dlt/dlt_common.h
+++ b/include/dlt/dlt_common.h
@@ -377,9 +377,9 @@ enum {
  * Provision to test static function
  */
 #ifndef DLT_UNIT_TESTS
-#define STATIC static
+#define DLT_STATIC static
 #else
-#define STATIC
+#define DLT_STATIC
 #endif
 /**
  * The definition of the serial header containing the characters "DLS" + 0x01.

--- a/src/daemon/dlt_daemon_connection.c
+++ b/src/daemon/dlt_daemon_connection.c
@@ -64,7 +64,7 @@ static DltConnectionId connectionId;
  * @return The amount of bytes send on success, -1 otherwise.
  *         errno is appropriately set.
  */
-STATIC int dlt_connection_send(DltConnection *conn,
+DLT_STATIC int dlt_connection_send(DltConnection *conn,
                               void *msg,
                               size_t msg_size)
 {
@@ -158,7 +158,7 @@ DltConnection *dlt_connection_get_next(DltConnection *current, int type_mask)
     return current;
 }
 
-STATIC void dlt_connection_destroy_receiver(DltConnection *con)
+DLT_STATIC void dlt_connection_destroy_receiver(DltConnection *con)
 {
     if (!con)
         return;
@@ -190,7 +190,7 @@ STATIC void dlt_connection_destroy_receiver(DltConnection *con)
  *
  * @return DltReceiver structure or NULL if none corresponds to the type.
  */
-STATIC DltReceiver *dlt_connection_get_receiver(DltDaemonLocal *daemon_local,
+DLT_STATIC DltReceiver *dlt_connection_get_receiver(DltDaemonLocal *daemon_local,
                                                 DltConnectionType type,
                                                 int fd)
 {

--- a/src/daemon/dlt_daemon_event_handler.c
+++ b/src/daemon/dlt_daemon_event_handler.c
@@ -341,7 +341,7 @@ DltConnection *dlt_event_handler_find_connection(DltEventHandler *ev,
  *
  * @return 0 on success, -1 if the connection is not found.
  */
-STATIC int dlt_daemon_remove_connection(DltEventHandler *ev,
+DLT_STATIC int dlt_daemon_remove_connection(DltEventHandler *ev,
                                        DltConnection *to_remove)
 {
     if (ev == NULL || to_remove == NULL)
@@ -417,7 +417,7 @@ void dlt_event_handler_cleanup_connections(DltEventHandler *ev)
  * @param ev The event handler structure where the connection list is.
  * @param connection The connection to be added.
  */
-STATIC void dlt_daemon_add_connection(DltEventHandler *ev,
+DLT_STATIC void dlt_daemon_add_connection(DltEventHandler *ev,
                                      DltConnection *connection)
 {
 

--- a/src/daemon/dlt_daemon_offline_logstorage.c
+++ b/src/daemon/dlt_daemon_offline_logstorage.c
@@ -41,7 +41,7 @@
  * @param ctxid          Context id as .* stored here
  * @return               0 on success -1 on error
  */
-STATIC DltReturnValue dlt_logstorage_split_ecuid(char *key,
+DLT_STATIC DltReturnValue dlt_logstorage_split_ecuid(char *key,
                                                int len,
                                                char *ecuid,
                                                char *appid,
@@ -70,7 +70,7 @@ STATIC DltReturnValue dlt_logstorage_split_ecuid(char *key,
  * @param ctxid          Context id from key stored here
  * @return               0 on success -1 on error
  */
-STATIC DltReturnValue dlt_logstorage_split_ctid(char *key,
+DLT_STATIC DltReturnValue dlt_logstorage_split_ctid(char *key,
                                               int len,
                                               char *appid,
                                               char *ctxid)
@@ -97,7 +97,7 @@ STATIC DltReturnValue dlt_logstorage_split_ctid(char *key,
  * @param ctxid          Context id as .* stored here
  * @return               0 on success -1 on error
  */
-STATIC DltReturnValue dlt_logstorage_split_apid(char *key,
+DLT_STATIC DltReturnValue dlt_logstorage_split_apid(char *key,
                                               int len,
                                               char *appid,
                                               char *ctxid)
@@ -124,7 +124,7 @@ STATIC DltReturnValue dlt_logstorage_split_apid(char *key,
  * @param ctxid          CContext id from key is stored here
  * @return               0 on success -1 on error
  */
-STATIC DltReturnValue dlt_logstorage_split_apid_ctid(char *key,
+DLT_STATIC DltReturnValue dlt_logstorage_split_apid_ctid(char *key,
                                               int len,
                                               char *appid,
                                               char *ctxid)
@@ -172,7 +172,7 @@ STATIC DltReturnValue dlt_logstorage_split_apid_ctid(char *key,
  * @param ctxid          CContext id as .* stored here
  * @return               0 on success -1 on error
  */
-STATIC DltReturnValue dlt_logstorage_split_ecuid_apid(char *key,
+DLT_STATIC DltReturnValue dlt_logstorage_split_ecuid_apid(char *key,
                                                int len,
                                                char *ecuid,
                                                char *appid,
@@ -222,7 +222,7 @@ STATIC DltReturnValue dlt_logstorage_split_ecuid_apid(char *key,
  * @param key            Prepared key stored here
  * @return               None
  */
-STATIC DltReturnValue dlt_logstorage_split_multi(char *key,
+DLT_STATIC DltReturnValue dlt_logstorage_split_multi(char *key,
                                            int len,
                                            char *ecuid,
                                            char *appid,
@@ -278,7 +278,7 @@ STATIC DltReturnValue dlt_logstorage_split_multi(char *key,
  * @param ctxid    Context id
  * @return         0 on success, -1 on error
  */
-STATIC DltReturnValue dlt_logstorage_split_key(char *key, char *appid, char *ctxid, char *ecuid)
+DLT_STATIC DltReturnValue dlt_logstorage_split_key(char *key, char *appid, char *ctxid, char *ecuid)
 {
     int len = 0;
     char *sep = NULL;
@@ -339,7 +339,7 @@ STATIC DltReturnValue dlt_logstorage_split_key(char *key, char *appid, char *ctx
  * @param loglevel      requested log level
  * @param verbose       verbosity flag
  */
-STATIC DltReturnValue dlt_daemon_logstorage_update_passive_node_context(
+DLT_STATIC DltReturnValue dlt_daemon_logstorage_update_passive_node_context(
         DltDaemonLocal *daemon_local,
         char *apid,
         char *ctid,
@@ -404,7 +404,7 @@ STATIC DltReturnValue dlt_daemon_logstorage_update_passive_node_context(
  * @param verbose           If set to true verbose information is printed out
  * @return                  0 on success, -1 on error
  */
-STATIC DltReturnValue dlt_daemon_logstorage_send_log_level(DltDaemon *daemon,
+DLT_STATIC DltReturnValue dlt_daemon_logstorage_send_log_level(DltDaemon *daemon,
                                                 DltDaemonLocal *daemon_local,
                                                 DltDaemonContext *context,
                                                 char *ecuid,
@@ -474,7 +474,7 @@ STATIC DltReturnValue dlt_daemon_logstorage_send_log_level(DltDaemon *daemon,
  * @param verbose           If set to true verbose information is printed out
  * @return                  0 on success, -1 on error
  */
-STATIC DltReturnValue dlt_daemon_logstorage_reset_log_level(DltDaemon *daemon,
+DLT_STATIC DltReturnValue dlt_daemon_logstorage_reset_log_level(DltDaemon *daemon,
                                                  DltDaemonLocal *daemon_local,
                                                  DltDaemonContext *context,
                                                  char *ecuid,
@@ -529,7 +529,7 @@ STATIC DltReturnValue dlt_daemon_logstorage_reset_log_level(DltDaemon *daemon,
  * @param verbose           If set to true verbose information is printed out
  * @return                  0 on success, -1 on error
  */
-STATIC DltReturnValue dlt_daemon_logstorage_force_reset_level(DltDaemon *daemon,
+DLT_STATIC DltReturnValue dlt_daemon_logstorage_force_reset_level(DltDaemon *daemon,
                                                    DltDaemonLocal *daemon_local,
                                                    char *apid,
                                                    char *ctxid,

--- a/src/gateway/dlt_gateway.c
+++ b/src/gateway/dlt_gateway.c
@@ -72,7 +72,7 @@ typedef enum {
  * @param value string to be tested
  * @return 0 on success, -1 otherwise
  */
-STATIC DltReturnValue dlt_gateway_check_ip(DltGatewayConnection *con, char *value)
+DLT_STATIC DltReturnValue dlt_gateway_check_ip(DltGatewayConnection *con, char *value)
 {
     struct sockaddr_in sa;
     int ret = DLT_RETURN_ERROR;
@@ -113,7 +113,7 @@ STATIC DltReturnValue dlt_gateway_check_ip(DltGatewayConnection *con, char *valu
  * @param value   string to be tested
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_gateway_check_port(DltGatewayConnection *con, char *value)
+DLT_STATIC int dlt_gateway_check_port(DltGatewayConnection *con, char *value)
 {
     int tmp = -1;
 
@@ -146,7 +146,7 @@ STATIC int dlt_gateway_check_port(DltGatewayConnection *con, char *value)
  * @param value   string to be used as ECU identifier
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_gateway_check_ecu(DltGatewayConnection *con, char *value)
+DLT_STATIC int dlt_gateway_check_ecu(DltGatewayConnection *con, char *value)
 {
     if ((con == NULL) || (value == NULL))
     {
@@ -171,7 +171,7 @@ STATIC int dlt_gateway_check_ecu(DltGatewayConnection *con, char *value)
  * @param value   string to be tested
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_gateway_check_connect_trigger(DltGatewayConnection *con,
+DLT_STATIC int dlt_gateway_check_connect_trigger(DltGatewayConnection *con,
                                              char *value)
 {
     if ((con == NULL) || (value == NULL))
@@ -205,7 +205,7 @@ STATIC int dlt_gateway_check_connect_trigger(DltGatewayConnection *con,
  * @param value   string to be tested
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_gateway_check_timeout(DltGatewayConnection *con, char *value)
+DLT_STATIC int dlt_gateway_check_timeout(DltGatewayConnection *con, char *value)
 {
     if ((con == NULL) || (value == NULL))
     {
@@ -230,7 +230,7 @@ STATIC int dlt_gateway_check_timeout(DltGatewayConnection *con, char *value)
  * @param value string to be tested
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_gateway_check_send_serial(DltGatewayConnection *con, char *value)
+DLT_STATIC int dlt_gateway_check_send_serial(DltGatewayConnection *con, char *value)
 {
     if ((con == NULL) || (value == NULL))
     {
@@ -249,7 +249,7 @@ STATIC int dlt_gateway_check_send_serial(DltGatewayConnection *con, char *value)
  * @param con   DltGatewayConnection to be updated
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_gateway_allocate_control_messages(DltGatewayConnection *con)
+DLT_STATIC int dlt_gateway_allocate_control_messages(DltGatewayConnection *con)
 {
     if (con == NULL)
     {
@@ -292,7 +292,7 @@ STATIC int dlt_gateway_allocate_control_messages(DltGatewayConnection *con)
  * @param value string to be tested
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_gateway_check_control_messages(DltGatewayConnection *con,
+DLT_STATIC int dlt_gateway_check_control_messages(DltGatewayConnection *con,
                                               char *value)
 {
     /* list of allowed clients given */
@@ -367,7 +367,7 @@ STATIC int dlt_gateway_check_control_messages(DltGatewayConnection *con,
  * @param value string to be tested
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_gateway_check_periodic_control_messages(DltGatewayConnection *con,
+DLT_STATIC int dlt_gateway_check_periodic_control_messages(DltGatewayConnection *con,
                                               char *value)
 {
     char *token = NULL;
@@ -501,7 +501,7 @@ STATIC int dlt_gateway_check_periodic_control_messages(DltGatewayConnection *con
  * Caution: after changing entries here,
  * dlt_gateway_check_param needs to be updated as well
  * */
-STATIC DltGatewayConf configuration_entries[GW_CONF_COUNT] =
+DLT_STATIC DltGatewayConf configuration_entries[GW_CONF_COUNT] =
 {
     [GW_CONF_IP_ADDRESS] = {
         .key = "IPaddress",
@@ -548,7 +548,7 @@ STATIC DltGatewayConf configuration_entries[GW_CONF_COUNT] =
  * @param value specified property value from configuration file
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_gateway_check_param(DltGateway *gateway,
+DLT_STATIC int dlt_gateway_check_param(DltGateway *gateway,
                                    DltGatewayConnection *con,
                                    DltGatewayConfType ctype,
                                    char *value)
@@ -849,7 +849,7 @@ void dlt_gateway_deinit(DltGateway *gateway, int verbose)
  * @param verbose       verbose flag
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_gateway_add_to_event_loop(DltDaemonLocal *daemon_local,
+DLT_STATIC int dlt_gateway_add_to_event_loop(DltDaemonLocal *daemon_local,
                                          DltGatewayConnection *con,
                                          int verbose)
 {
@@ -1050,7 +1050,7 @@ DltReceiver *dlt_gateway_get_connection_receiver(DltGateway *gateway, int fd)
  * @param verbose verbose flag
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_gateway_parse_get_log_info(DltDaemon *daemon,
+DLT_STATIC int dlt_gateway_parse_get_log_info(DltDaemon *daemon,
                                           char *ecu,
                                           DltMessage *msg,
                                           int req,
@@ -1189,7 +1189,7 @@ STATIC int dlt_gateway_parse_get_log_info(DltDaemon *daemon,
  * @param verbose verbose flag
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_gateway_parse_get_default_log_level(DltDaemon *daemon,
+DLT_STATIC int dlt_gateway_parse_get_default_log_level(DltDaemon *daemon,
                                                    DltDaemonLocal *daemon_local,
                                                    char *ecu,
                                                    DltMessage *msg,
@@ -1241,7 +1241,7 @@ STATIC int dlt_gateway_parse_get_default_log_level(DltDaemon *daemon,
  * @param verbose      int
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_gateway_control_service_logstorage(DltDaemon *daemon,
+DLT_STATIC int dlt_gateway_control_service_logstorage(DltDaemon *daemon,
                                                   DltDaemonLocal *daemon_local,
                                                   int verbose)
 {

--- a/src/lib/dlt_client.c
+++ b/src/lib/dlt_client.c
@@ -978,7 +978,7 @@ int dlt_client_set_socket_path(DltClient *client, char *socket_path)
  * @param resp          DltServiceGetLogInfoResponse
  * @param count_app_ids number of app_ids which needs to be freed
  */
-STATIC void dlt_client_free_calloc_failed_get_log_info(DltServiceGetLogInfoResponse *resp,
+DLT_STATIC void dlt_client_free_calloc_failed_get_log_info(DltServiceGetLogInfoResponse *resp,
                                                       int count_app_ids)
 {
     AppIDsType *app = NULL;

--- a/src/offlinelogstorage/dlt_offline_logstorage.c
+++ b/src/offlinelogstorage/dlt_offline_logstorage.c
@@ -71,7 +71,7 @@ typedef enum {
     DLT_LOGSTORAGE_FILTER_CONF_COUNT
 } DltLogstorageFilterConfType;
 
-STATIC void dlt_logstorage_filter_config_free(DltLogStorageFilterConfig *data)
+DLT_STATIC void dlt_logstorage_filter_config_free(DltLogStorageFilterConfig *data)
 {
     DltLogStorageFileList *n = NULL;
     DltLogStorageFileList *n1 = NULL;
@@ -113,7 +113,7 @@ STATIC void dlt_logstorage_filter_config_free(DltLogStorageFilterConfig *data)
     }
 }
 
-STATIC int dlt_logstorage_list_destroy(DltLogStorageFilterList **list,
+DLT_STATIC int dlt_logstorage_list_destroy(DltLogStorageFilterList **list,
                                            int reason)
 {
     DltLogStorageFilterList *tmp = NULL;
@@ -150,7 +150,7 @@ STATIC int dlt_logstorage_list_destroy(DltLogStorageFilterList **list,
     return 0;
 }
 
-STATIC int dlt_logstorage_list_add_config(DltLogStorageFilterConfig *data,
+DLT_STATIC int dlt_logstorage_list_add_config(DltLogStorageFilterConfig *data,
                                               DltLogStorageFilterConfig **listdata)
 {
     if (*(listdata) == NULL)
@@ -192,7 +192,7 @@ STATIC int dlt_logstorage_list_add_config(DltLogStorageFilterConfig *data,
     return 0;
 }
 
-STATIC int dlt_logstorage_list_add(char *key,
+DLT_STATIC int dlt_logstorage_list_add(char *key,
                                        DltLogStorageFilterConfig *data,
                                        DltLogStorageFilterList **list)
 {
@@ -245,7 +245,7 @@ STATIC int dlt_logstorage_list_add(char *key,
     return 0;
 }
 
-STATIC void *dlt_logstorage_list_find(char *key,
+DLT_STATIC void *dlt_logstorage_list_find(char *key,
                                           DltLogStorageFilterList **list)
 {
     while (*(list) != NULL)
@@ -265,7 +265,7 @@ STATIC void *dlt_logstorage_list_find(char *key,
 
 /* Configuration file parsing helper functions */
 
-STATIC int dlt_logstorage_count_ids(const char *str)
+DLT_STATIC int dlt_logstorage_count_ids(const char *str)
 {
 
     if (str == NULL)
@@ -327,7 +327,7 @@ void dlt_logstorage_free(DltLogStorage *handle, int reason)
  * @param value        string given in config file
  * @return             0 on success, -1 on error
  */
-STATIC int dlt_logstorage_read_list_of_names(char **names, char *value)
+DLT_STATIC int dlt_logstorage_read_list_of_names(char **names, char *value)
 {
     int i = 0;
     int y = 0;
@@ -402,7 +402,7 @@ STATIC int dlt_logstorage_read_list_of_names(char **names, char *value)
  * @param value        string given in config file
  * @return             0 on success, -1 on error
  */
-STATIC int dlt_logstorage_read_number(unsigned int *number, char *value)
+DLT_STATIC int dlt_logstorage_read_number(unsigned int *number, char *value)
 {
     int i = 0;
     int len = 0;
@@ -451,7 +451,7 @@ STATIC int dlt_logstorage_read_number(unsigned int *number, char *value)
  * @param numids         Number of keys in the list is stored here
  * @return: 0 on success, error on failure*
  */
-STATIC int dlt_logstorage_get_keys_list(char *ids, char *sep, char **list,
+DLT_STATIC int dlt_logstorage_get_keys_list(char *ids, char *sep, char **list,
                                         int *numids)
 {
     char *token = NULL;
@@ -510,7 +510,7 @@ STATIC int dlt_logstorage_get_keys_list(char *ids, char *sep, char **list,
  * @param key            Prepared key stored here
  * @return               None
  */
-STATIC void dlt_logstorage_create_keys_only_ctid(char *ecuid, char *ctid,
+DLT_STATIC void dlt_logstorage_create_keys_only_ctid(char *ecuid, char *ctid,
                                                  char *key)
 {
     char curr_str[DLT_OFFLINE_LOGSTORAGE_MAX_KEY_LEN + 1] = { 0 };
@@ -543,7 +543,7 @@ STATIC void dlt_logstorage_create_keys_only_ctid(char *ecuid, char *ctid,
  * @param key            Prepared key stored here
  * @return               None
  */
-STATIC void dlt_logstorage_create_keys_only_apid(char *ecuid, char *apid,
+DLT_STATIC void dlt_logstorage_create_keys_only_apid(char *ecuid, char *apid,
                                                  char *key)
 {
     char curr_str[DLT_OFFLINE_LOGSTORAGE_MAX_KEY_LEN + 1] = { 0 };
@@ -578,7 +578,7 @@ STATIC void dlt_logstorage_create_keys_only_apid(char *ecuid, char *apid,
  * @param key            Prepared key stored here
  * @return               None
  */
-STATIC void dlt_logstorage_create_keys_multi(char *ecuid, char *apid,
+DLT_STATIC void dlt_logstorage_create_keys_multi(char *ecuid, char *apid,
                                              char *ctid, char *key)
 {
     char curr_str[DLT_OFFLINE_LOGSTORAGE_MAX_KEY_LEN + 1] = { 0 };
@@ -613,7 +613,7 @@ STATIC void dlt_logstorage_create_keys_multi(char *ecuid, char *apid,
  * @param key            Prepared key stored here
  * @return               None
  */
-STATIC void dlt_logstorage_create_keys_only_ecu(char *ecuid, char *key)
+DLT_STATIC void dlt_logstorage_create_keys_only_ecu(char *ecuid, char *key)
 {
     char curr_str[DLT_OFFLINE_LOGSTORAGE_MAX_KEY_LEN + 1] = { 0 };
 
@@ -650,7 +650,7 @@ STATIC void dlt_logstorage_create_keys_only_ecu(char *ecuid, char *key)
  * @param[out]: num_keys: number of keys
  * @return: 0 on success, error on failure*
  */
-STATIC int dlt_logstorage_create_keys(char *apids,
+DLT_STATIC int dlt_logstorage_create_keys(char *apids,
                                       char *ctids,
                                       char *ecuid,
                                       char **keys,
@@ -753,7 +753,7 @@ STATIC int dlt_logstorage_create_keys(char *apids,
  * @param tmp_data       Holds all other configuration values
  * @return               0 on success, -1 on error
  */
-STATIC int dlt_logstorage_prepare_table(DltLogStorage *handle,
+DLT_STATIC int dlt_logstorage_prepare_table(DltLogStorage *handle,
                                         DltLogStorageFilterConfig *data)
 {
     int ret = 0;
@@ -812,7 +812,7 @@ STATIC int dlt_logstorage_prepare_table(DltLogStorage *handle,
  * @return               0 on success, -1 on error
  *
  */
-STATIC int dlt_logstorage_validate_filter_name(char *name)
+DLT_STATIC int dlt_logstorage_validate_filter_name(char *name)
 {
     int len = 0;
     int idx = 0;
@@ -875,7 +875,7 @@ STATIC int dlt_logstorage_validate_filter_name(char *name)
     }
 }
 
-STATIC void dlt_logstorage_filter_set_strategy(DltLogStorageFilterConfig *config,
+DLT_STATIC void dlt_logstorage_filter_set_strategy(DltLogStorageFilterConfig *config,
                                                int strategy)
 {
     if (config == NULL)
@@ -899,7 +899,7 @@ STATIC void dlt_logstorage_filter_set_strategy(DltLogStorageFilterConfig *config
     }
 }
 
-STATIC int dlt_logstorage_check_apids(DltLogStorageFilterConfig *config,
+DLT_STATIC int dlt_logstorage_check_apids(DltLogStorageFilterConfig *config,
                                       char *value)
 {
     if ((config == NULL) || (value == NULL))
@@ -911,7 +911,7 @@ STATIC int dlt_logstorage_check_apids(DltLogStorageFilterConfig *config,
     return dlt_logstorage_read_list_of_names(&config->apids, value);
 }
 
-STATIC int dlt_logstorage_check_ctids(DltLogStorageFilterConfig *config,
+DLT_STATIC int dlt_logstorage_check_ctids(DltLogStorageFilterConfig *config,
                                       char *value)
 {
     if ((config == NULL) || (value == NULL))
@@ -922,7 +922,7 @@ STATIC int dlt_logstorage_check_ctids(DltLogStorageFilterConfig *config,
     return dlt_logstorage_read_list_of_names(&config->ctids, value);
 }
 
-STATIC int dlt_logstorage_check_loglevel(DltLogStorageFilterConfig *config,
+DLT_STATIC int dlt_logstorage_check_loglevel(DltLogStorageFilterConfig *config,
                                          char *value)
 {
     if ((config == NULL) || (value == NULL))
@@ -970,7 +970,7 @@ STATIC int dlt_logstorage_check_loglevel(DltLogStorageFilterConfig *config,
     return 0;
 }
 
-STATIC int dlt_logstorage_check_reset_loglevel(DltLogStorageFilterConfig *config,
+DLT_STATIC int dlt_logstorage_check_reset_loglevel(DltLogStorageFilterConfig *config,
                                                char *value)
 {
     if (config == NULL)
@@ -1022,7 +1022,7 @@ STATIC int dlt_logstorage_check_reset_loglevel(DltLogStorageFilterConfig *config
     return 0;
 }
 
-STATIC int dlt_logstorage_check_filename(DltLogStorageFilterConfig *config,
+DLT_STATIC int dlt_logstorage_check_filename(DltLogStorageFilterConfig *config,
                                          char *value)
 {
     int len;
@@ -1062,7 +1062,7 @@ STATIC int dlt_logstorage_check_filename(DltLogStorageFilterConfig *config,
     return 0;
 }
 
-STATIC int dlt_logstorage_check_filesize(DltLogStorageFilterConfig *config,
+DLT_STATIC int dlt_logstorage_check_filesize(DltLogStorageFilterConfig *config,
                                          char *value)
 {
     if ((config == NULL) || (value == NULL))
@@ -1073,7 +1073,7 @@ STATIC int dlt_logstorage_check_filesize(DltLogStorageFilterConfig *config,
     return dlt_logstorage_read_number(&config->file_size, value);
 }
 
-STATIC int dlt_logstorage_check_nofiles(DltLogStorageFilterConfig *config,
+DLT_STATIC int dlt_logstorage_check_nofiles(DltLogStorageFilterConfig *config,
                                         char *value)
 {
     if ((config == NULL) || (value == NULL))
@@ -1084,7 +1084,7 @@ STATIC int dlt_logstorage_check_nofiles(DltLogStorageFilterConfig *config,
     return dlt_logstorage_read_number(&config->num_files, value);
 }
 
-STATIC int dlt_logstorage_check_specificsize(DltLogStorageFilterConfig *config,
+DLT_STATIC int dlt_logstorage_check_specificsize(DltLogStorageFilterConfig *config,
                                              char *value)
 {
     if ((config == NULL) || (value == NULL))
@@ -1106,7 +1106,7 @@ STATIC int dlt_logstorage_check_specificsize(DltLogStorageFilterConfig *config,
  * @param value        string given in config file
  * @return             0 on success, -1 on error
  */
-STATIC int dlt_logstorage_check_sync_strategy(DltLogStorageFilterConfig *config,
+DLT_STATIC int dlt_logstorage_check_sync_strategy(DltLogStorageFilterConfig *config,
                                               char *value)
 {
     if ((config == NULL) || (value == NULL))
@@ -1167,7 +1167,7 @@ STATIC int dlt_logstorage_check_sync_strategy(DltLogStorageFilterConfig *config,
  * @param value        string given in config file
  * @return             0 on success, -1 on error
  */
-STATIC int dlt_logstorage_check_ecuid(DltLogStorageFilterConfig *config,
+DLT_STATIC int dlt_logstorage_check_ecuid(DltLogStorageFilterConfig *config,
                                       char *value)
 {
     int len;
@@ -1194,7 +1194,7 @@ STATIC int dlt_logstorage_check_ecuid(DltLogStorageFilterConfig *config,
     return 0;
 }
 
-STATIC DltLogstorageFilterConf
+DLT_STATIC DltLogstorageFilterConf
     filter_cfg_entries[DLT_LOGSTORAGE_FILTER_CONF_COUNT] = {
     [DLT_LOGSTORAGE_FILTER_CONF_LOGAPPNAME] = {
         .key = "LogAppName",
@@ -1249,7 +1249,7 @@ STATIC DltLogstorageFilterConf
 };
 
 /* */
-STATIC DltLogstorageFilterConf
+DLT_STATIC DltLogstorageFilterConf
     filter_nonverbose_storage_entries[DLT_LOGSTORAGE_FILTER_CONF_COUNT] = {
     [DLT_LOGSTORAGE_FILTER_CONF_LOGAPPNAME] = {
         .key = NULL,
@@ -1303,7 +1303,7 @@ STATIC DltLogstorageFilterConf
     }
 };
 
-STATIC DltLogstorageFilterConf
+DLT_STATIC DltLogstorageFilterConf
 filter_nonverbose_control_entries[DLT_LOGSTORAGE_FILTER_CONF_COUNT] = {
     [DLT_LOGSTORAGE_FILTER_CONF_LOGAPPNAME] = {
         .key = "LogAppName",
@@ -1364,7 +1364,7 @@ filter_nonverbose_control_entries[DLT_LOGSTORAGE_FILTER_CONF_COUNT] = {
  * @param value specified property value from configuration file
  * @return 0 on success, -1 otherwise
  */
-STATIC int dlt_logstorage_check_param(DltLogStorageFilterConfig *config,
+DLT_STATIC int dlt_logstorage_check_param(DltLogStorageFilterConfig *config,
                                       DltLogstorageFilterConfType ctype,
                                       char *value)
 {
@@ -1381,8 +1381,7 @@ STATIC int dlt_logstorage_check_param(DltLogStorageFilterConfig *config,
     return -1;
 }
 
-
-STATIC int dlt_logstorage_get_filter_value(DltConfigFile *config_file,
+DLT_STATIC int dlt_logstorage_get_filter_value(DltConfigFile *config_file,
                                            char *sec_name,
                                            int index,
                                            char *value)
@@ -1479,8 +1478,8 @@ STATIC int dlt_logstorage_get_filter_value(DltConfigFile *config_file,
     return 0;
 }
 
-STATIC int dlt_logstorage_setup_table(DltLogStorage *handle,
-                                      DltLogStorageFilterConfig *tmp_data)
+DLT_STATIC int dlt_logstorage_setup_table(DltLogStorage *handle,
+                                          DltLogStorageFilterConfig *tmp_data)
 {
     int ret = 0;
 
@@ -1503,7 +1502,7 @@ STATIC int dlt_logstorage_setup_table(DltLogStorage *handle,
    DLT_OFFLINE_LOGSTORAGE_STORE_FILTER_ERROR - On error while storing in hash table
  */
 
-STATIC int dlt_daemon_offline_setup_filter_properties(DltLogStorage *handle,
+DLT_STATIC int dlt_daemon_offline_setup_filter_properties(DltLogStorage *handle,
                                               DltConfigFile *config_file,
                                               char *sec_name)
 {
@@ -1602,7 +1601,7 @@ STATIC int dlt_daemon_offline_setup_filter_properties(DltLogStorage *handle,
  * @return                   0 on success, -1 on error, 1 on warning
  *
  */
-STATIC int dlt_logstorage_store_filters(DltLogStorage *handle,
+DLT_STATIC int dlt_logstorage_store_filters(DltLogStorage *handle,
                                         char *config_file_name)
 {
     DltConfigFile *config = NULL;
@@ -1696,7 +1695,7 @@ STATIC int dlt_logstorage_store_filters(DltLogStorage *handle,
  * @param input_file    pointer to configuration file stored on device
  * @return              0 on success, -1 on error, 1 on warning
  */
-STATIC int dlt_logstorage_load_config(DltLogStorage *handle)
+DLT_STATIC int dlt_logstorage_load_config(DltLogStorage *handle)
 {
     char config_file_name[PATH_MAX + 1] = {'\0'};
     int ret = 0;
@@ -2009,7 +2008,7 @@ int dlt_logstorage_get_config(DltLogStorage *handle,
  * @param ecuid     EcuID given in the message
  * @return          number of found configurations
  */
-STATIC int dlt_logstorage_filter(DltLogStorage *handle,
+DLT_STATIC int dlt_logstorage_filter(DltLogStorage *handle,
                                  DltLogStorageFilterConfig **config,
                                  char *apid,
                                  char *ctid,

--- a/src/offlinelogstorage/dlt_offline_logstorage_behavior.c
+++ b/src/offlinelogstorage/dlt_offline_logstorage_behavior.c
@@ -564,9 +564,9 @@ int dlt_logstorage_open_log_file(DltLogStorageFilterConfig *config,
  * @param cnt         count
  * @return index on success, -1 on error
  */
-STATIC int dlt_logstorage_find_dlt_header(void *ptr,
-                                          unsigned int offset,
-                                          unsigned int cnt)
+DLT_STATIC int dlt_logstorage_find_dlt_header(void *ptr,
+                                              unsigned int offset,
+                                              unsigned int cnt)
 {
     int index = 0;
     char substring[] = {'D', 'L', 'T', 0x01};
@@ -593,8 +593,8 @@ STATIC int dlt_logstorage_find_dlt_header(void *ptr,
  * @param config      DltLogStorageFilterConfig
  * @param ret         return value of fwrite call
  */
-STATIC void dlt_logstorage_check_write_ret(DltLogStorageFilterConfig *config,
-                                           int ret)
+DLT_STATIC void dlt_logstorage_check_write_ret(DltLogStorageFilterConfig *config,
+                                               int ret)
 {
     if (config == NULL)
     {
@@ -632,11 +632,11 @@ STATIC void dlt_logstorage_check_write_ret(DltLogStorageFilterConfig *config,
  * @param remain_file_size  log file remaining size
  * @return 0 on success, -1 on error
  */
-STATIC DltReturnValue dlt_logstorage_sync_create_new_file(
-    DltLogStorageFilterConfig *config,
-    DltLogStorageUserConfig *file_config,
-    char *dev_path,
-    unsigned int remain_file_size)
+DLT_STATIC DltReturnValue dlt_logstorage_sync_create_new_file(
+        DltLogStorageFilterConfig *config,
+        DltLogStorageUserConfig *file_config,
+        char *dev_path,
+        unsigned int remain_file_size)
 {
     int index = 0;
     int ret;
@@ -750,10 +750,10 @@ STATIC DltReturnValue dlt_logstorage_sync_create_new_file(
  * @param dev_path      Storage device mount point path
  * @return 0 on success, -1 on error
  */
-STATIC DltReturnValue dlt_logstorage_sync_to_file(
-    DltLogStorageFilterConfig *config,
-    DltLogStorageUserConfig *file_config,
-    char *dev_path)
+DLT_STATIC DltReturnValue dlt_logstorage_sync_to_file(
+        DltLogStorageFilterConfig *config,
+        DltLogStorageUserConfig *file_config,
+        char *dev_path)
 {
     int ret = 0;
     unsigned int remain_file_size = 0;
@@ -816,9 +816,9 @@ STATIC DltReturnValue dlt_logstorage_sync_to_file(
  * @param index_status  check for index is required or not
  * @return 0 on success, -1 on error
  */
-STATIC DltReturnValue dlt_logstorage_sync_capable_data_to_file(
-    DltLogStorageFilterConfig *config,
-    int index_status)
+DLT_STATIC DltReturnValue dlt_logstorage_sync_capable_data_to_file(
+        DltLogStorageFilterConfig *config,
+        int index_status)
 {
     int ret = 0;
     int index = 0;

